### PR TITLE
Pagination for trades

### LINKED ISSUE #3653: Implement backend paginations for trades and deposits/withdrawals
## Problem Definition

To improve frontend performance we need to move the pagination/filtering/sorting for the trades along with the deposits withdrawals to the backend.

And then adjust the frontend to use the pagination (similar to how transactions work).

### DEX Trades
At the moment the frontend will fetch both CEX trades and DEX trades then it will transform the DEX trades to the same schema format as CEX trades and merge the results.

If we move the pagination/filtering/sorting to the backend we probably need to make sure that the `GET /api/(version)/trades` endpoint can also return DEX trades in the CEX trade schema format, and also make these trade locations available when filtering by location.

### Filter Suggestions
The frontend will read some of the fields and will suggest values based on the search/keyword for autocompletion.

For the trades, this includes the quote/base asset and the location,
and for the deposits & withdrawals, this includes the asset and location.

The current implementation uses the actual entries in memory to generate these suggestions. With the pagination/filtering/sorting moving to the backend, there are two different approaches we can take.

To ensure that the suggestions match real data a new endpoint would need to be implemented.

Alternatively, we can just use all the trade locations and available assets as a suggestion source. This doesn't need a new endpoint and when the users select a suggestion that doesn't have data they might see an empty result set.

### PR DESCRIPTION
**Pull request by LefterisJP**
Pagination for trades

Backend part of trades for #3653

### CODE REVIEW

**Comment by LefterisJP** [comment]
All green :)

**Comment by kelsos** [comment]
@lukicenturi it seems that using the `dev-proxy` is not required anymore since the PR is merged.

To get a basic idea of the changes for this you should see how the transactions work with the server-based pagination.

The main point is that the pagination/filtering/sorting for trades will now happen in the backend, the frontend should only keep a page in memory.

All the filtering/sorting needs to happen in a sync query instead of an `async_query=true` request that uses `only_cache=true` to make the db the single source of truth for the frontend.

docs are [here](https://rotki.readthedocs.io/en/latest/api.html#get--api-(version)-trades)

The `async_query=true` requests should now be used only to update/refresh the data instead.

We should prioritize this work over other front-end issues.

**Comment by LefterisJP** [comment]
And same would go for deposit/withdrawals and ledger actions ... as soon as I manage to do the work for them too :rofl:

File: rotkehlchen/api/v1/encoding.py [hunk 1]
```diff
@@ -969,6 +986,58 @@ def make_ethereum_transaction_query(  # pylint: disable=no-self-use
...
+    location = LocationField(load_default=None)
+
+    @validates_schema
+    def validate_trades_query_schema(  # pylint: disable=no-self-use
+            self,
+            data: Dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        value = data['order_by_attribute']
+        if data['order_by_attribute'] not in (None, 'time'):
```
Comment (lukicenturi) [inline rotkehlchen/api/v1/encoding.py:1009]: shouldn't this has to be `timestamp`? @LefterisJP

Comment (lukicenturi) [reply rotkehlchen/api/v1/encoding.py:1009]: Nevermind, it's supposed to be `time` I think. I will convert it in frontend

Comment (LefterisJP) [reply rotkehlchen/api/v1/encoding.py:1009]: answered you in discord

**Pull request merged**